### PR TITLE
Link to printf config in platform config page

### PR DESCRIPTION
**Pull request by kegilbert**
@AnotherButler 

Sorry for the back to back PR. This came up while working on the platform page.

### CODE REVIEW

**Comment by kegilbert** [comment]
The standing issue blocking this is there are two config options for baud rate in platform at the moment:
`default-serial-baud-rate`
`stdio-baud-rate`

The second one is fine in platform, but the first one should be in drivers as it affects Serial objects default baud rate. So as it stands my new sentence is not correct. Working on a way to move that config option. If it proofs too much effort I'll rephrase the sentence.

File: docs/reference/configuration/Drivers.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <h2 id="configuration-drivers">Drivers</h2>
 
-The driver configurations allows for customization of the OS driver layer. Options in this section pertain to hardware IO. 
+The driver configurations allows for customization of the OS driver layer. Options in this section pertain to hardware IO. You can find additional serial settings for `printf` in the [platform page](platform.html). These settings do not affect Serial objects.
```
Comment (AnotherButler) [inline docs/reference/configuration/Drivers.md]: Note: This currently links to the Platform overview page of the user API section. Is that correct, or did you want to link to the platform config page (configuration-platform.html)?

**Comment by AnotherButler** [comment]
@kegilbert @SenRamakri Is this ready for merging?

File: docs/reference/configuration/Drivers.md (hunk 1 above)
Comment (kegilbert) [reply docs/reference/configuration/Drivers.md]: Nice catch, configuration-platform is correct.

**Comment by kegilbert** [comment]
@AnotherButler 

Haven't gotten a lot of feedback on this since it's pretty low priority: https://github.com/ARMmbed/mbed-os/issues/7265

I just updated the blurb to mention the exception, I think it's fine to merge now: 443774a3f5448bdba2db50127df2a35de8bbe983

**Pull request merged**